js_modules/dagster-ui/packages/ui-core/src/assets/catalog/AssetsGroupedView.tsx

```diff
@@ -0,0 +1,540 @@
+            assets={assets}
+            icon={<Icon name={propertyToLabelAndIcon[selectedTab].icon} size={16} />}
+            link={link}
+          />
+        ),
+      );
+  },
+);
+
+const List = ({rows}: {rows: React.ReactNode[]}) => {
+  const scrollWrapperRef = useRef<HTMLDivElement | null>(null);
+  const rowVirtualizer = useVirtualizer({
+    count: rows.length,
+    getScrollElement: () => scrollWrapperRef.current,
+    estimateSize: () => 28,
+    overscan: 5,
+  });
+
+  const rowItems = rowVirtualizer.getVirtualItems();
+  const totalHeight = rowVirtualizer.getTotalSize();
+
+  return (
+    <Container ref={scrollWrapperRef} style={{maxHeight: '600px', overflowY: 'auto'}}>
+      <Inner $totalHeight={totalHeight}>
+        {rowItems.map(({index, key, size, start}) => {
+          const row = rows[index]!;
+          return (
+            <Row key={key} $height={size} $start={start}>
+              <div ref={rowVirtualizer.measureElement} data-index={index}>
+                {row}
+              </div>
+            </Row>
+          );
+        })}
+      </Inner>
+    </Container>
+  );
+};
+
+const propertyToLabelAndIcon: Record<Property, {label: string; icon: IconName}> = {
+  groupName: {
+    label: 'Groups',
+    icon: 'asset_group',
+  },
+  kinds: {
+    label: 'Kinds',
+    icon: 'compute_kind',
+  },
+  owners: {
+    label: 'Owners',
+    icon: 'account_circle',
+  },
+  tags: {
+    label: 'Tags',
+    icon: 'tag',
+  },
+  repository: {
+    label: 'Code locations',
+    icon: 'folder',
+  },
+};
+
+// DRY: Shared Section Header for Selections
+const SelectionSectionHeader = ({
+  icon,
+  label,
+  count,
+  border,
+  isOpen,
+  toggleOpen,
+  children,
+  displayAs,
+}: {
+  icon: IconName;
+  label: string;
+  count: number;
+  border: 'top-and-bottom' | 'bottom';
+  isOpen: boolean;
+  toggleOpen: () => void;
+  children?: React.ReactNode;
+  displayAs: 'List' | 'Grid';
+}) => {
+  const actuallyOpen = useDeferredValue(isOpen);
+  return (
+    <Box padding={{bottom: actuallyOpen && displayAs === 'Grid' ? 24 : 0}}>
+      <Box
+        flex={{
+          direction: 'row',
+          alignItems: 'center',
+          gap: 8,
+          justifyContent: 'space-between',
+        }}
+        padding={{horizontal: 24, vertical: 2}}
+        style={{height: 44}}
+        background={Colors.backgroundLight()}
+        border={border}
+      >
+        <Box flex={{direction: 'row', alignItems: 'center', gap: 8}}>
+          <Icon name={icon} />
+          <Subtitle>
+            {label} ({numberFormatter.format(count)})
+          </Subtitle>
+          <UnstyledButton
+            onClick={(e) => {
+              e.stopPropagation();
+              toggleOpen();
+            }}
+            onDoubleClick={(e) => {
+              e.stopPropagation();
+            }}
+            onKeyDown={(e) => {
+              if (e.code === 'Space') {
+                e.preventDefault();
+              }
+            }}
```

Comment:
What's the outcome when you don't stop propagation on these?

Reply:
the row's link gets clicked

Reply:
Yeah, that's what I was thinking, just wanted to verify.